internal/statestore/public.go

```diff
@@ -36,14 +36,18 @@ type Service interface {
 	// DeleteTicket removes the Ticket with the specified id from state storage. This method succeeds if the Ticket does not exist.
 	DeleteTicket(ctx context.Context, id string) error
 
-	// IndexTicket indexes the Ticket id for the configured index fields.
+	// IndexTicket adds the ticket to the index.
 	IndexTicket(ctx context.Context, ticket *pb.Ticket) error
 
-	// DeindexTicket removes the indexing for the specified Ticket. Only the indexes are removed but the Ticket continues to exist.
+	// DeindexTicket removes specified ticket from the index. The Ticket continues to exist.
 	DeindexTicket(ctx context.Context, id string) error
 
-	// FilterTickets returns the Ticket ids for the Tickets meeting the specified filtering criteria.
-	FilterTickets(ctx context.Context, pool *pb.Pool, pageSize int, callback func([]*pb.Ticket) error) error
+	// GetIndexedIds returns the ids of all tickets currently indexed.
+	GetIndexedIds(ctx context.Context) (map[string]struct{}, error)
+
+	// GetTickets returns multiple tickets from storage.  Missing tickets are
+	// silently ignored.
+	GetTickets(ctx context.Context, ids []string) ([]*pb.Ticket, error)
```

Comment:
Can we combine the interface of `GetTickets` and `GetTicket` to something like `GetTickets(ctx context.Context, ids ...string)`? Having two functions solving the same things seems redundant to me. @sawagh what do you think?

Reply:
Ah yeah. Unless the single one has any optimizations over the multiple one - we can just do with GetTickets

Reply:
I anticipate there being a difference in the future: whether assignments are included or not.  This is if we only combine assignments on query time. see #1137 for my rambling.  Otherwise these could be combined in the future.